odxtools/modelyear.py

```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+from xml.etree import ElementTree
+
+from .infocomponent import InfoComponent
+from .odxdoccontext import OdxDocContext
+from .utils import dataclass_fields_asdict
+
+
+@dataclass(kw_only=True)
+class ModelYear(InfoComponent):
```

Comment:
This looks wrong, sub-classing InfoComponent each time, without added value

Reply:
the added value is that `isinstance()` can be used to distinguish these objects and thus it is much clearer to ensure that stuff like `ECU-CONFIG-REF` points to an `EcuConfig` object. (I thought about using type aliases like the derivatives of `ValidityFor`, but IMO using a separate class is better for the mentioned reasons and it should be considered to change the `ValidityFor` family to use separate classes as well.)